tests/unit/test_units.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+import numpy.testing as npt
+import pytest
+from scmdata.testing import get_single_ts
+
+from local.h2_adjust.units import _get_unit_scaling, sanitize_production_intensity_units
+
+
+@pytest.mark.parametrize(
+    "source,target,exp",
+    [
+        ("kg NOx / MWh", "kg NOx / MWh", 1),
+        ("kg NOx / MWh", "t NOx / MWh", 1 / 1000),
+        ("g NOx/kWh", "kg NOx / MWh", 1),
+        ("kg NOx/TJ", "kg NOx / MWh", 3600 / 1e6),
+        ("kg /t", "kg / kg", 1 / 1000),
+    ],
+)
+def test_convert_intensities(source, target, exp):
+    np.testing.assert_almost_equal(_get_unit_scaling(source, target), exp)
+
+
+@pytest.mark.parametrize(
+    "unit,carrier,product,exp",
+    (
+        ("% of H2", "H2", "H2", 0.01),
+        ("kgNOx/tNH3", "NH3", "NOx", 1 / 1000 * (14 + 3) / 3),
+        ("kgNH3/tNH3", "NH3", "NH3", 1 / 1000 * (14 + 3) / 3),
+        ("% H2 component of fuel", "NH3", "H2", 0.01),
+        ("% H2 component of fuel", "CH4", "H2", 0.01),
```

Comment:
In this case we are trying to calculate `kg {product} / kg H" = "kg H / kg H". Hence the 1/100 factor (% to fraction). 

The NH3 carrier doesn't matter here because of where this factor is used. In the 210 notebook these factors are used to convert the mass of H2 for a given sector/carrier into mass of emissions for a product. We don't convert the mass H2 into the equivalent mass of carrier as I don't think we need to and it can be rather difficult (e.g. how to go from kg H2 -> kg synFuel).

Reply:
> In this case we are trying to calculate `kg {product} / kg H" = "kg H / kg H".

In this case we are going from kg emissions of gas per kg of hydrogen to kg H per kg H? I'm still confused about what this unit is saying? Can you try and phrase something like, "The unit '% H2 component of fuel' means that [for each tonne of fuel that is burnt, we assume that 1% of that mass is released as H2]."? Hopefully the penny will then drop, if not we can jump on zoom.

Reply:
The later. These are leakage rates of H2 from pipelines etc. 

My notation wasn't that clear - the product in this particular case is H2 so we are trying to derive kg H2 / kg H2 (or kg H/kg H). In this particular case the carrier plays no role in the derivation.

RBs notes on this particular row:
` Leakage from transport of H2 to point of recombination for facilities not co-located. From Frazer-Nash 50th CI estimates`

Reply:
> In this particular case the carrier plays no role in the derivation

Ah ok that's what had confused me (context is important here and I didn't get it, my bad, but also hard to carry in context in the units). Looks all good now. I put a suggestion above that might help (but might just not be much we can do)